src/jit/codegenarmarch.cpp

```diff
@@ -1617,11 +1766,29 @@ void CodeGen::genCallInstruction(GenTreeCall* call)
         }
         else
         {
-            regNumber argReg = curArgTabEntry->regNum;
             genConsumeReg(argNode);
```

Comment:
I cannot understand why we need to update the liveness when `genConsumeReg()` is called from `genCallInstruction()`. I thought that we should update liveness in `genConsumeReg()` when the argument `tree` of `genConsumeReg()` represents local variable (In other words, `tree->OperIsLocal() == true`). 
When `genConsumeReg()` is called from `genCallInstruction()`, `tree` does not represent local variable (it always has `GT_PUTARG_REG` as oper in general, and `GT_PUTARG_SPLIT` for split struct on arm32). As a result, updating livenesss in `genConsumeReg()` is skipped. So  I modified only `genUnspillRegIfNeeded()`.
Please tell me what I misunderstand.